packages/vite-plugin-checker/src/checkers/eslint/main.ts

```diff
@@ -111,15 +111,20 @@ export class EslintChecker extends Checker<'eslint'> {
         buildBin: (pluginConfig) => {
           let ext = ['.js']
           let files: string[] = []
+          let maxWarnings: number | undefined
+
           if (pluginConfig.eslint) {
             ext = pluginConfig.eslint.extensions ?? ext
+            maxWarnings = pluginConfig.eslint.maxWarnings
             files =
               typeof pluginConfig.eslint.files === 'string'
                 ? [pluginConfig.eslint.files]
                 : pluginConfig.eslint.files
           }
 
-          return ['eslint', ['--ext', ext.join(','), ...files]]
+          maxWarnings ??= -1
+
+          return ['eslint', ['--ext', ext.join(','), `--max-warnings=${maxWarnings}`, ...files]]
```

Comment:
Have to use `=` here, as `eslint --max-warnings -1` fails because it tries to parse `-1` as a flag.